src/components/encoding-pane/property-editor.tsx

```diff
@@ -14,24 +19,61 @@ export interface PropertyEditorProps extends ActionHandler<SpecEncodingAction> {
   fieldDef: ShelfFieldDef;
 }
 
+export interface PropertyEditorSchema {
+  uiSchema: any;
+  schema: any;
+}
+
 export class PropertyEditorBase extends React.PureComponent<PropertyEditorProps, {}> {
+  private static getPropertyEditorSchema(prop: string, nestedProp: string): PropertyEditorSchema {
+    if (prop === 'scale') {
+      if (nestedProp === 'type') {
+        return {
+          schema: SCALE_TYPE_SCHEMA,
```

Comment:
I think `SCALE_TYPE_SCHEMA` is actually used only once.  We can stick the return value in here, esp. if we move this function to the other file already?